secretstores/azure/keyvault/authutils.go

```diff
@@ -17,15 +17,35 @@ import (
 	"golang.org/x/crypto/pkcs12"
 )
 
-// ClientAuthorizer provides the options to get a bearer authorizer from a client certificate.
-type ClientAuthorizer struct {
+// CertConfig provides the options to get a bearer authorizer from a client certificate.
+type CertConfig struct {
 	*auth.ClientCertificateConfig
 	CertificateData []byte
 }
 
-// NewClientAuthorizer creates an ClientAuthorizer object configured to obtain an Authorizer through Client Credentials.
-func NewClientAuthorizer(certificatePath string, certificateBytes []byte, certificatePassword string, clientID string, tenantID string) ClientAuthorizer {
-	return ClientAuthorizer{
+// GetClientCert creates a config object from the available certificate credentials.
+// An error is returned if no certificate credentials are available.
+func (k *keyvaultSecretStore) GetClientCert() (CertConfig, error) {
```

Comment:
Change the utils method then to take the new settings struct.
like GetClientCert(setting EnvironmentSettings) ...

Reply:
Solved in commit 8225aae.

Thanks for the review, this is a cleaner solution.